### PR TITLE
Revert "Replace deprecated onReportTimings w/ frameTimings"

### PR DESCRIPTION
**Pull request by liyuqian**
Reverts flutter/flutter#38861

Reason for revert: Stream is considered a bad API: https://github.com/flutter/engine/pull/11041#issuecomment-526892247

### CODE REVIEW

**Comment by devoncarew** [comment]
I'm not clear what the issues with the Stream API are. Do we have a proposal for what would replace the API? The issues with the previous API where that clients of the API had to manage ownership of a field, and would likely mostly do that unsuccessfully.

**Comment by liyuqian** [comment]
@devoncarew : yes, I'm thinking about storing the callbacks as a list like [addPostFrameCallback](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/scheduler/binding.dart#L605). @Hixie : does that sound good to you?

**Comment by hpoul** [comment]
I'm also curious about the problems with streams? it doesn't seem too long ago that BLoC + stream was the go-to solution.. is there some major difference between framework use of streams, vs. framework users? I tend to follow the lead of the framework to have some uniform solution vs. having everything doing things differently (callbacks vs Listenable/ChangeNotifier vs stream).

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by dnfield** [comment]
@hixie created another section in the style guide for this here: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#use-of-streams-in-flutter-framework-code

**Comment by Hixie** [comment]
Streams are fine in general, they're just not the best solution to this problem.

File: packages/flutter/lib/src/widgets/binding.dart [hunk 1]
```diff
@@ -747,13 +747,25 @@ mixin WidgetsBinding on BindingBase, SchedulerBinding, GestureBinding, RendererB
...
         }
+        if (oldCallback != null) {
+          oldCallback(timings);
+        }
+        // use frameTimings. https://github.com/flutter/flutter/issues/38838
+        // ignore: deprecated_member_use
+        WidgetsBinding.instance.window.onReportTimings = oldCallback;
         _firstFrameCompleter.complete();
-      });
+      };
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/binding.dart]: this all probably belongs in the SchedulerBinding, fwiw.

**Pull request merged**